subsys/usb/device/usb_device.c

```diff
@@ -570,7 +570,7 @@ static bool set_endpoint(const struct usb_ep_descriptor *ep_desc)
 	if (ep_bm & usb_dev.ep_bm) {
 		reset_endpoint(ep_desc);
 		/* allow any canceled transfers to terminate */
-		k_usleep(150);
+		k_busy_wait(150);
```

Comment:
To block here is contrary to the statement "allow any canceled transfers to terminate".
The workaround for STM32F shim driver would probably to skip it in ISR context:
```suggestion
		if (!k_is_in_isr()) {
			k_usleep(150);
		}
```